### PR TITLE
Shorten file names

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
We're shortening the file names here to avoid running into max path problems.

### CODE REVIEW

File: src/DI.Specification.Tests/DI.Specification.Tests.csproj [hunk 1]
```diff
@@ -5,6 +5,8 @@
   <PropertyGroup>
     <Description>Suite of xUnit.net tests to check for container compatibility with Microsoft.Extensions.DependencyInjection.</Description>
     <TargetFramework>netstandard2.0</TargetFramework>
+    <AssemblyName>Microsoft.Extensions.DependencyInjection.Specification.Tests</AssemblyName>
+    <RootNamespace>Microsoft.Extensions.DependencyInjection.Specification</RootNamespace>
```
Comment (Eilon) [inline src/DI.Specification.Tests/DI.Specification.Tests.csproj]: Any reason to have this deviate? Normally if these two settings aren't set, their default value is the project's filename or something, no?

File: test/DI.Tests/DI.Tests.csproj [hunk 2]
```diff
@@ -5,11 +5,13 @@
   <PropertyGroup>
     <TargetFrameworks>netcoreapp2.0;net461</TargetFrameworks>
     <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netcoreapp2.0</TargetFrameworks>
+    <AssemblyName>Microsoft.Extensions.DependencyInjection.Tests</AssemblyName>
+    <RootNamespace>Microsoft.Extensions.DependencyInjection</RootNamespace>
```
Comment (Eilon) [inline test/DI.Tests/DI.Tests.csproj]: See previous comment.

File: src/DI.Specification.Tests/DI.Specification.Tests.csproj (hunk 1 above)
Comment (ryanbrandenburg) [inline src/DI.Specification.Tests/DI.Specification.Tests.csproj]: Don't we not want namespaces in test repos to include `*.Tests`? Or is that not what this property means?

Comment (ryanbrandenburg) [inline src/DI.Specification.Tests/DI.Specification.Tests.csproj]: Oh, this is in `src` not `test`. Still, all the namespaces in this project are `Microsoft.Extensions.DependencyInjection.Specification` not `Microsoft.Extensions.DependencyInjection.Specification.Tests`.

**Comment by ryanbrandenburg** [comment]
@Eilon and I talked through the RootNamespace stuff offline. We decided that it should be safe to do it this way provided there are no RESX files in the project, and so long as we check that the resultant DLL's are the same.

For posterity: The benefit of having different AssemblyName and RootNamespace is that when someone does "Add new *.cs" they get a file with the correct namespace (no `.Tests`) by default.

**Comment by pakrym** [comment]
Do we really want to call it DI? Isn't `DependencyInjection` enough?

**Comment by natemcmaster** [comment]
DI is too short. Would it be okay to keep the folders like this?
```
DependencyInjection
DependencyInjection.Abstractions
DependencyInjection.Specification.Tests
```

**Comment by Eilon** [comment]
Why is DI too short? The container folder is already `DependencyInjection/src/`. We're needing to save every possible character here...

**Comment by pakrym** [comment]
Just feels like overacronymization

**Comment by Eilon** [comment]
If you have alternative ideas on how to solve this, that'd be great. But we need to do as much as we can to shorten names.

**Comment by natemcmaster** [comment]
The only alternative idea I have is what I suggested above. My concern is consistency. What will the pattern be on other repos? Should we be using acronyms as we do this? e.g. does "Microsoft.Extensions.Configuration/" become "CFG/"?

Also, @ryanbrandenburg are you doing this across all repos?

**Comment by Eilon** [comment]
`Microsoft.Extensions.Configuration` could easily become `Config`. I think that's both very readable and understandable.

**Pull request merged**